src/graphnet/models/gnn/RNN_tito.py

```diff
@@ -0,0 +1,125 @@
+            global_pooling_schemes (Union[str, List[str]], optional): Pooling schemes to use. Defaults to None.
+            embedding_dim (int, optional): Embedding dimension of the RNN. Defaults to None ie. no embedding.
+            n_head (int, optional): Number of heads to use in the DynTrans model. Defaults to 16.
+            use_global_features (bool, optional): Whether to use global features after pooling. Defaults to True.
+            use_post_processing_layers (bool, optional): Whether to use post-processing layers after the DynTrans layers. Defaults to True.
+        """
+        self._nb_neighbours = nb_neighbours
+        self._nb_inputs = nb_inputs
+        self._RNN_layers = RNN_layers
+        self._RNN_hidden_size = RNN_hidden_size  # RNN_hidden_size
+        self._RNN_dropout = RNN_dropout
+        self._embedding_dim = embedding_dim
+        self._n_head = n_head
+        self._use_global_features = use_global_features
+        self._use_post_processing_layers = use_post_processing_layers
+
+        self._features_subset = features_subset
+        if dyntrans_layer_sizes is None:
+            dyntrans_layer_sizes = [
+                (256, 256),
+                (256, 256),
+                (256, 256),
+                (256, 256),
+            ]
+        else:
+            dyntrans_layer_sizes = [
+                tuple(layer_sizes) for layer_sizes in dyntrans_layer_sizes
+            ]
+
+        self._dyntrans_layer_sizes = dyntrans_layer_sizes
+        self._post_processing_layer_sizes = post_processing_layer_sizes
+        self._global_pooling_schemes = global_pooling_schemes
+        if readout_layer_sizes is None:
+            readout_layer_sizes = [
+                256,
+                128,
+            ]
+        self._readout_layer_sizes = readout_layer_sizes
+
+        super().__init__(nb_inputs, self._readout_layer_sizes[-1])
+
+        self._rnn = Node_RNN(
+            num_layers=self._RNN_layers,
+            nb_inputs=2,
+            hidden_size=self._RNN_hidden_size,
+            RNN_dropout=self._RNN_dropout,
+            embedding_dim=self._embedding_dim,
+        )
+
+        self._dynedge_tito = DynEdgeTITO(
+            nb_inputs=self._RNN_hidden_size + 5,
+            dyntrans_layer_sizes=self._dyntrans_layer_sizes,
+            features_subset=self._features_subset,
+            global_pooling_schemes=self._global_pooling_schemes,
+            use_global_features=self._use_global_features,
+            use_post_processing_layers=self._use_post_processing_layers,
+            post_processing_layer_sizes=self._post_processing_layer_sizes,
+            readout_layer_sizes=self._readout_layer_sizes,
+            n_head=self._n_head,
+            nb_neighbours=self._nb_neighbours,
+        )
+
+    def forward(self, data: Data) -> torch.Tensor:
+        """Apply learnable forward pass of the RNN and tito model."""
+        data = self._rnn(data)
+        readout = self._dynedge_tito(data)
```

Comment:
`self._dynedge_tito` could in principle be any of our existing backbones, right?

Reply:
yes.